msgpack/_unpacker.pyx

```diff
@@ -150,6 +195,8 @@ def unpackb(object packed, object object_hook=None, object list_hook=None,
              use_list, cenc, cerr,
              max_str_len, max_bin_len, max_array_len, max_map_len, max_ext_len)
     ret = unpack_construct(&ctx, buf, buf_len, &off)
+    if new_protocol:
+        PyBuffer_Release(&view);
```

Comment:
try ~ finally is required, since unpack_construct, encoding.encode() and unicode_errors.encode() may raise Exception.